dependencyManagement/build.gradle.kts

```diff
@@ -33,8 +36,16 @@ val slf4jVersion = "2.0.16"
 val opencensusVersion = "0.31.1"
 val prometheusClientVersion = "0.16.0"
 val prometheusServerVersion = "1.3.5"
+val armeriaVersion = "1.31.3"
+
 
 val DEPENDENCIES = listOf(
+  "org.junit.jupiter:junit-jupiter-api:5.11.4",
```

Comment:
Can we follow the armeria pattern and explicitly add dependency versions here for each of the junit components we use? I.e.:
```
"org.junit.jupiter:junit-jupiter-api:${junitVersion}",
"org.junit.jupiter:junit-jupiter-params:${junitVersion}",
"org.junit.jupiter:junit-jupiter-pioneer:${junitVersion}",
```

If we can, we should try to retain the pattern of establishing all dependency versions here and not letting the junit dependency version leak into `otel-java-conventions.gradle.kts`.